haystack/nodes/retriever/base.py

```diff
@@ -296,7 +296,7 @@ def run_batch(  # type: ignore
         headers: Optional[Dict[str, str]] = None,
     ):
         if root_node == "Query":
-            if not queries:
+            if queries is None:
                 raise HaystackError(
                     "Must provide a 'queries' parameter for retrievers in pipelines where Query is the root node."
                 )
```

Comment:
Same as above (untested code):

```suggestion
            if queries is None:
                raise HaystackError(
                    "Must provide a 'queries' parameter for retrievers in pipelines where Query is the root node."
                )
            if not all(isinstance(query, str) for query in queries):
                logger.error("The retriever received an unusual list of queries: '%s' Some of these queries are likely to produce garbage output.", queries)
```

Reply:
Done.